include/wlr/types/wlr_xdg_shell.h

```diff
@@ -121,10 +127,16 @@ struct wlr_xdg_toplevel {
 	struct wlr_xdg_surface *parent;
 	struct wl_listener parent_unmap;
 
-	struct wlr_xdg_toplevel_state client_pending;
-	struct wlr_xdg_toplevel_state server_pending;
-	struct wlr_xdg_toplevel_state last_acked;
-	struct wlr_xdg_toplevel_state current;
+	struct wlr_xdg_toplevel_state current, pending;
+
+	// Properties to be sent to the client in the next configure event.
+	struct wlr_xdg_toplevel_configure scheduled;
+	struct wlr_xdg_toplevel_configure last_acked;
```

Comment:
Storing this `last_acked` configure is entirely redundant with the `pending` state. I think we should instead construct this configure on the stack from the `pending` state when needed for comparision in `compare_xdg_surface_toplevel_state()`.

Reply:
Makes sense. Will remove it a bit later.